### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by vazgen23h**
Homework 3 without Problem166 and chess desk.

### CODE REVIEW

File: homeworks/vazgen_harutyunyan/Homework_3/Problem151.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+				naturalNumbers[i]=i;
+			}
+			for(int i=0;i<naturalNumbers.length;i++){
+				if(naturalNumbers%n==0){
+					sum=sum+naturalNumbers;
+				}
+
+			}
+			System.out.println(sum);
+
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem151.java:25]: wrong indentation

File: homeworks/vazgen_harutyunyan/Homework_3/Problem151.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		int lastNumber; //array.length=lastNumber
+		int n;
+		int sum=0;
+		Scanner scanner=new Scanner(System.in);
+			System.out.println("Enter your Last Nuber");
+			lastNumber=scanner.nextInt();
+			System.out.println("Enter your test number");
+			n=scanner.nextInt();
+
+			int[] naturalNumbers =new int[lastNumber];
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem151.java:14]: why you need an array? Couldn't the sum calculated without it?

File: homeworks/vazgen_harutyunyan/Homework_3/Problem159.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        int firstNumber = 100;
+        int arrayLength = 900;
+        int[] array = new int[arrayLength];
+        for(int i = 0; i < 900; i++) {
+            array[i] = firstNumber;
+            firstNumber++;
+        }
+
+        for(int i = 0; i < 900; i++) {
+            if(array[i] % 3 == 1 || array[i]%4==2 ) {
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem159.java]: the logical and should be used instead and it should also solve the out of range problem

File: homeworks/vazgen_harutyunyan/Homework_3/Problem162.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Task162
+// Find the largest four-digit number that multiplied by 14 becomes the square of a natural number.
+public class Problem162 {
+    public static void main(String[] args) {
+        int firstNumber = 1000;
+        int firstArrayLength = 9000;
+        int max = 0;
+        int[] firstArray = new int[firstArrayLength];
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem162.java:8]: you dont need array to keep natural number, for indices can be used instead

File: homeworks/vazgen_harutyunyan/Homework_3/Problem171.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        int n;
+        int factorial=1;
+        Scanner scanner = new Scanner(System.in);
+        n=scanner.nextInt();
+        if(n==0){
+        	System.out.print(factorial);
+        }
+        for(int i=1;i<=n;i++){
+        	factorial=factorial*i;
+        }
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem171.java:14]: this approach should have been used for all cases, you did not need to use arrays

File: homeworks/vazgen_harutyunyan/Homework_3/Problem174.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import java.util.Scanner;
+public class Problem174 {
+    public static void main(String[] args) {
+        double a = 2;
+        double b = 2 + (1 / a);
+        int n;
+        Scanner scanner = new Scanner(System.in);
+        n = scanner.nextInt();
+        for(int i = 0; i < n; i++) {
```
Comment (sirak92) [inline homeworks/vazgen_harutyunyan/Homework_3/Problem174.java:9]: this is the right coding style for "for" loop, meaning the whitespaces

**Review by sirak92 — COMMENTED** [review]
In general these group of tasks would not consider using arrays.

File: homeworks/vazgen_harutyunyan/Homework_3/Problem171.java (hunk 5 above)
Comment (vazgen23h) [reply homeworks/vazgen_harutyunyan/Homework_3/Problem171.java:14]: But in this task, I didn't use an array.

File: homeworks/vazgen_harutyunyan/Homework_3/Problem159.java (hunk 3 above)
Comment (vazgen23h) [reply homeworks/vazgen_harutyunyan/Homework_3/Problem159.java]: But where will I get out of range problem? My 3 digit numbers start from 100 to 999. There are (999-100)+1 numbers. So my arraylength=900. Then I start to check my if statement.

**Pull request merged**